packages/core/src/utils/zod.ts

```diff
@@ -41,10 +119,20 @@ export const zodTypeToSwagger = (config: unknown): OpenAPIV3.SchemaObject => {
     };
   }
 
+  if (config instanceof ZodLiteral) {
+    return zodLiteralToSwagger(config);
+  }
+
   if (config instanceof ZodUnknown) {
     return { example: {} }; // Any data type
   }
 
+  if (config instanceof ZodUnion) {
+    return {
+      oneOf: (config.options as ZodType[]).map((option) => zodTypeToSwagger(option)),
```

Comment:
nit: What's the inferred type for `config.options`? Can we remove the `as ZodType[]` and check the type in `zodTypeToSwagger`?

Reply:
if we remove `as ZodType[]`, we cannot infer its actual type.

since we can know the actual type according to the source code, I decide to use `as ZodType[]`.

<img width="939" alt="image" src="https://user-images.githubusercontent.com/10594507/174014269-a6dada11-0123-45cd-ac79-f21ddc1f8d3e.png">
<img width="808" alt="image" src="https://user-images.githubusercontent.com/10594507/174014307-a85ac976-71f3-4c54-9a52-9ff1a477831a.png">
<img width="628" alt="image" src="https://user-images.githubusercontent.com/10594507/174014349-511b2c7d-6b58-4df0-b027-859f3525555f.png">

any better idea? @logto-io/eng